airflow/utils/log/s3_task_handler.py

```diff
@@ -169,5 +173,7 @@ def s3_write(self, log, remote_log_location, append=True):
                 replace=True,
                 encrypt=configuration.conf.getboolean('core', 'ENCRYPT_S3_LOGS'),
             )
+            return True
         except Exception:
             self.log.exception('Could not write logs to %s', remote_log_location)
+        return False
```

Comment:
I would leave the details of local logs deleting to other folks as I haven't touched that part before.

But I do have a question about adding the True/False return statement. Seems it's only for the tests' convenience? If that's the case, possibly they're not really necessary. Instead, it can be handled in the tests themselves directly.

Reply:
Hey XD, while the boolean makes testing easier, I added the True/False return to serve as a flag for successful deletion. Right now, if there's an issue uploading the logs to S3 during the `s3_write` method, the failure is logged out, but no exception is thrown. To prevent logs from being deleted when they haven't been successfully uploaded, I added the boolean to make the upload's status explicit.